staging/src/k8s.io/component-base/logs/json/register/register_test.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package register
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/component-base/config"
+	"k8s.io/component-base/logs"
+)
+
+func TestJSONFlag(t *testing.T) {
+	o := logs.NewOptions()
+	fs := pflag.NewFlagSet("addflagstest", pflag.ContinueOnError)
+	output := bytes.Buffer{}
+	o.AddFlags(fs)
+	fs.SetOutput(&output)
+	fs.PrintDefaults()
+	want := `      --experimental-logging-sanitization   [Experimental] When enabled prevents logging of fields tagged as sensitive (passwords, keys, tokens).
+                                            Runtime log sanitization may introduce significant computation overhead and therefore should not be enabled in production.
+      --logging-format string               Sets the log format. Permitted formats: "json", "text".
```

Comment:
Nit: We should only validate that "json" is available in sub string  `Permitted formats: "json", "text".`. Verifying more means that every time we rename a flag/update description we will need to update multiple places which is not great.

Reply:
+1 to this

Reply:
This PR https://github.com/kubernetes/kubernetes/pull/103173 to change. Thanks for catch.